api/v1/inferencepool_types.go

```diff
@@ -58,21 +62,21 @@ type InferencePoolSpec struct {
 	// If specified, it will be applied to match the model server pods in the same namespace as the InferencePool.
 	// Cross namesoace selector is not supported.
 	//
-	// +kubebuilder:validation:Required
-	Selector map[LabelKey]LabelValue `json:"selector"`
+	// +required
+	Selector map[LabelKey]LabelValue `json:"selector,omitempty"`
 
 	// TargetPortNumber defines the port number to access the selected model server Pods.
 	// The number must be in the range 1 to 65535.
 	//
 	// +kubebuilder:validation:Minimum=1
 	// +kubebuilder:validation:Maximum=65535
-	// +kubebuilder:validation:Required
-	TargetPortNumber int32 `json:"targetPortNumber"`
+	// +required
+	TargetPortNumber int32 `json:"targetPortNumber,omitempty"`
```

Comment:
Required fields should not contain `omitempty`.

Reply:
I made a similar comment on Ricardo's PR on Gateway API, but apparently API conventions have recently changed here: https://github.com/kubernetes-sigs/gateway-api/pull/3905#discussion_r2191414197

Reply:
https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#serialization-of-optionalrequired-fields